### PR TITLE
Adding container based ACA unit test with build for cross platform development

### PR DESCRIPTION
**Pull request by liangbin-pub**
- To build in container: build.sh
- To build and run unit test: build.sh test

Note: No need to have OVS support in host, tested on Mac OsX and Linux host

Test Result:
[----------] Global test environment tear-down
[==========] 30 tests from 5 test suites ran. (17572 ms total)
[  PASSED  ] 30 tests.

  YOU HAVE 5 DISABLED TESTS

g_total_network_configuration_time = 45712663156 nanoseconds or 45712 milliseconds
g_total_update_GS_time = 4743972082 nanoseconds or 4743 milliseconds
Program exiting, cleaning up...
ACA_OVS_L2_Programmer::execute_openflow_command ---> Entering
Executing command: ovs-ofctl del-flows br-int "udp,udp_src=68,udp_dst=67"
Command succeeded!
 Elapsed time for system command took: 93502520 nanoseconds or 93 milliseconds.
Elapsed time for openflow client call took: 93765360 nanoseconds or 93 milliseconds. rc: 0
ACA_OVS_L2_Programmer::execute_openflow_command <--- Exiting, rc = 0
(NOT USED) Serialized protobuf string: "$27330ae4-b718-11ea-b3de-111111111111:$99d9d709-8478-4b46-9f3f-000000000000B$99d9d709-8478-4b46-9f3f-000000000000R
                                                                                                                                                          10.0.0.0/24X{b	10.10.0.1"fa:16:3e:d7:f2:11*�2$99d9d709-8478-4b46-9f3f-000000000000:$1b08a5bc-b718-11ea-b3de-111122223333B	portname1Jfa:16:3e:d7:f2:6cR
111.0.0.11Z1$27330ae4-b718-11ea-b3de-111111111111	11.0.0.11
(USING THIS) Serialized protobuf binary array: "$27330ae4-b718-11ea-b3de-111111111111:$99d9d709-8478-4b46-9f3f-000000000000B$99d9d709-8478-4b46-9f3f-000000000000R
                                                                                                                                                                  10.0.0.0/24X{b	10.10.0.1"fa:16:3e:d7:f2:11*�2$99d9d709-8478-4b46-9f3f-000000000000:$1b08a5bc-b718-11ea-b3de-111122223333B	portname1Jfa:16:3e:d7:f2:6cR
111.0.0.11Z1$27330ae4-b718-11ea-b3de-111111111111	11.0.0.11
Deserialize succeeded, comparing the content now...
All content matched!

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
+    -- add-port br-int patch-tun \
+    -- set interface patch-tun type=patch options:peer=patch-int \
+    -- add-port br-tun patch-int \
+    -- set interface patch-int type=patch options:peer=patch-tun && \
+    ovs-ofctl add-flow br-tun "table=0, priority=1,in_port="patch-int" actions=resubmit(,2)" && \
+    ovs-ofctl add-flow br-tun "table=2, priority=0 actions=resubmit(,22)"'
+
+  echo "    --- running alcor-control-agent ---"
+  # sends output to null device, but stderr to console 
+  docker exec a1 bash -c "nohup /mnt/host/code/build/bin/AlcorControlAgent -d > /dev/null 2>&1 &"
```
Comment (er1cthe0ne) [inline build/build.sh:52]: we may need to kill AlcorControlAgent when the tests are done.

**Review by er1cthe0ne — APPROVED** [review]
LGTM. Let me know when you are ready to merge and I will merge it.

File: build/build.sh (hunk 1 above)
Comment (liangbin-pub) [reply build/build.sh:52]: it is left there intentionally so user can do manual staff with docker exec inside container
The rebuild will delete the container instance, or user can delete it manually if not needed any more

**Comment by liangbin-pub** [comment]
good to merge

File: build/build.sh (hunk 1 above)
Comment (er1cthe0ne) [reply build/build.sh:52]: sounds good! Thanks.

**Pull request merged**